### ISSUE TITLE
Fix filetype detection given first line of file

### ISSUE THREAD

**Issue by sheerun** [sheerun/vim-polyglot#274]
As mentioned [here](https://github.com/sheerun/vim-polyglot/issues/262#issuecomment-356765211) we need to re-enable detecting filetype given fierst line of file, but in such way `getline` is called only once.

**Comment by sheerun** [comment]
It turns out default vim plugins have this already covered, mostly, so no action from vim-polyglot is necessary. If someone misses this for some filetype, please comment
